README.md

```diff
@@ -89,4 +89,20 @@ version topoSemverVersion("prefix")
 ```
 where prefix is the prefix for the tags you want to search (see Prefix tags).
 
-Topological sorting will then find the closest tag to HEAD that also has matches the prefix, and use that tag as the base for the version. 
\ No newline at end of file
+Topological sorting will then find the closest tag to HEAD that also has matches the prefix, and use that tag as the base for the version.
+
+## Version Object
+
+The plugin's version methods return a ``SemverVersion`` object and not a ``String``.  It can be used as a parameter for Gradle's `version`.  Making this an object allows for pulling out pieces of the version string for other uses (if desired).
+
+```java
+class SemverVersion {
+  String toString()  // The full version string
+  String getTagVersion() // The matched tagged
+  String getHeadCommitHash() // The git commit hash of the HEAD commit
+  Integer getCommitCount() // The number of commits HEAD is from the matched tag
+  Integer getBuildNumber() // The value of the BUILD_NUMBER environment variable
+  boolean isDirty() // Is the git repo dirty?
+  boolean isVersionStableRelease() // Does the full version represent a stable version?
```

Comment:
i'm not sure if we need this method on the object? its not really part of the semver version, more as part of the logic used when combining the parts into the final version string. It probably only really makes sense if calling toString() actually builds the version string using the parts (therefore it needs to know if its a stable release). See comment below